### ISSUE TITLE
Enhance options for feed downloads

### ISSUE THREAD

**Issue by vossenjp** [lwindolf/liferea#925]
https://blogs.oracle.com/linux/rss does not work in Liferea 1.12.2 (on Linux Mint-19.3).  Sometimes you get various parsing errors but other times it can totally hang Liferea and you have to kill it!  It's not just Liferea, neither `curl` nor `wget` _just work_ either.

You can get the feed to work using _command_ `curl --cookie-jar /tmp/oracle-cookies --user-agent 'Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:84.0) Gecko/20100101 Firefox/84.0' --max-time 20 https://blogs.oracle.com/linux/rss`.  I found both `--cookie-jar` and `--user-agent` to be required, but `--max-time` was optional.  I expected to need `--referer` but didn't.

So I propose adding "use cookies", "max time", and "change user agent" feed download options (see above `curl` examples).  "follow redirections" (`curl --location`) and "referer" (`curl --referer`) options would also be nice to have.

Obviously you can already do all of those things using `curl` but not everyone will know that or think of it, and having handy GUI options would be nice for handling over-complicated and badly designed RSS servers like Oracle's.

**Comment by lwindolf** [comment]
To be honest I'm a bit biased against fixing Liferea to be able to access Oracle feeds. I feel with you having to read what Oracle posts :-)

As a workaround please change your feed subscription to a command running exactly your curl statement!

**Comment by vossenjp** [comment]
Yeah, I debated about even creating this bug.  It took 5 minutes to write the bug and 20 minutes to remove all the OT rants about how much Oracle (the company & the DB) and the Unusable Enterprise Kernel (UEK) suck...  I ended up creating this because I figured:
1. Publishing the hack-around was not a bad idea, esp. if it might help for something else equally stupid but NOT-Oracle
2. Adding some more flexibility to the feed downloader might be worth at least talking about

I'd certainly label this as a low-priority maybe-nice-to-have.

**Comment by marciof** [comment]
I'm not sure if it's the same issue as OP has/had, but I've noticed Liferea sometimes seemingly not downloading a valid feed, only to confirm that `curl` does. I'm using version 1.13.5 on Ubuntu 21.04 64-bit.

Is there some debugging option or logging that can help root cause this issue?

**Comment by lwindolf** [comment]
@marciof you can run debugging using the command line option --debug-net this provides connection details and possible error codes.

**Comment by marciof** [comment]
@lwindolf, thank you. Here's what I found out. So Liferea actually does check the feed's URL, but it seems that when it gets a 304 (what I'm seeing now) it still runs the conversion filter, but with no data on stdin. Is this the expected behavior though?

**Comment by lwindolf** [comment]
@marciof While it might not be useful to run the filter on a HTTP 304 result, it should not hurt either as a 304 should have no content anyway. I guess if your feed doesn't parse it's caused in an earlier fetch request. If you could share the feed I could try reproducing. Could you add a new ticket for this?

**Comment by marciof** [comment]
@lwindolf, the feed does parse. I think I was just surprised to see the conversion filter still being invoked on empty content, as I didn't see any mention of this in the documentation.
